main.go

```diff
@@ -129,3 +156,45 @@ func vulnsBy(sev string, store map[string][]*clair.Vulnerability) []*clair.Vulne
 	}
 	return items
 }
+
+//Filter out whitelisted vulnerabilites
+func filterWhitelist(whitelist *vulnerabilitiesWhitelist, vs []*clair.Vulnerability) []*clair.Vulnerability {
+	generalWhitelist := (*whitelist).General
+	imageWhitelist := (*whitelist).Images
+	
+	filteredVs := make([]*clair.Vulnerability, 0, len(vs))
+	
+	for _, v := range vs {
+		if _, exists := generalWhitelist[v.Name]; !exists {
+			//vulnerability is not in the general whitelist, so get the image name by removing ":version" from the value returned via the Clair API
+			imageName := strings.Split(v.NamespaceName, ":")[0]
+			if _, exists := imageWhitelist[imageName][v.Name]; !exists {
+				//vulnerability is not in the image whitelist, so add it to the list to return
+				filteredVs = append(filteredVs, v)
+			}
+		}
+	}
+	
+	return filteredVs
+}
+
+//Remove duplicates
+func deDupe(vs []*clair.Vulnerability) []*clair.Vulnerability {
+	deDupedVs := make([]*clair.Vulnerability, 0, len(vs))
+	
+	//Use a map to store found vulnerabilities
+	foundVulnerabilites := make(map[string]bool)
+	
+	for _, v := range vs {
+		//use a combination of vulnerability name, feature name, and feature version to uniquely identify a vulnerability for deduping
+		if _, exists := foundVulnerabilites[ strings.Join([]string{v.Name,v.FeatureName,v.FeatureVersion},"") ]; !exists {
```

Comment:
since you generate "full name" twice I'd recommend to define a closure or function, also consider a different separator to prevent collision